core/types/confidential.go

```diff
@@ -89,6 +91,97 @@ func (tx *ConfidentialComputeRecord) setSignatureValues(chainID, v, r, s *big.In
 	tx.ChainID, tx.V, tx.R, tx.S = chainID, v, r, s
 }
 
+type ConfidentialComputeRequest2 struct {
```

Comment:
This is a transaction type, we could say that is the EIP-712 transaction type. Record is only the message type, like the Suave intent-format.